src/ssl.c

```diff
@@ -2947,15 +2947,15 @@ int wolfSSL_shutdown(WOLFSSL* ssl)
 
         /* call wolfSSL_shutdown again for bidirectional shutdown */
         if (ssl->options.sentNotify && !ssl->options.closeNotify) {
-            ret = wolfSSL_read(ssl, &tmp, 0);
+            ret = wolfSSL_read(ssl, &tmp, 1);
```

Comment:
This doesn't look right.  Actually pulling a byte of data off the secure stream?

Reply:
This read only occurs after the first shut down has been sent. By reading a byte, we can clear any pending app data so the close notify actually gets processed. This was a reproducible problem for a non-blocking application.